--- src.old/gcc/config/i386/t-mingw-w64	2010-10-24 22:40:46.235190117 +0200
+++ src/gcc/config/i386/t-mingw-w64	2010-10-24 23:04:48.769029001 +0200
@@ -1,12 +1,5 @@
 # Match SYSTEM_INCLUDE_DIR
 NATIVE_SYSTEM_HEADER_DIR = /include
 
-MULTILIB_OPTIONS = m64/m32
-MULTILIB_DIRNAMES = 64 32
-MULTILIB_OSDIRNAMES = ../lib64 ../lib32
-
 # MinGW-specific parts of LIB_SPEC
 SHLIB_LC = -lmingw32 -lmingwex -lmoldname -lmsvcrt -luser32 -lkernel32 -ladvapi32 -lshell32
-
-LIBGCC = stmp-multilib
-INSTALL_LIBGCC = install-multilib
