Description: Replace tabs with spaces in Ada code
Origin: http://gcc.gnu.org/viewcvs/branches/gcc-4_6-branch/gcc/ada/g-socthi-mingw.adb?r1=176306&r2=176305&view=patch&pathrev=176306
Bug-GNU: http://gcc.gnu.org/bugzilla/show_bug.cgi?id=48711

--- branches/gcc-4_6-branch/gcc/ada/g-socthi-mingw.adb	2011/07/15 05:52:04	176305
+++ branches/gcc-4_6-branch/gcc/ada/g-socthi-mingw.adb	2011/07/15 05:53:13	176306
@@ -277,8 +277,8 @@
       use type C.size_t;
 
       Fill  : constant Boolean :=
-	SOSC.MSG_WAITALL /= -1
-	and then (C.unsigned (Flags) and SOSC.MSG_WAITALL) /= 0;
+        SOSC.MSG_WAITALL /= -1
+          and then (C.unsigned (Flags) and SOSC.MSG_WAITALL) /= 0;
       --  Is the MSG_WAITALL flag set? If so we need to fully fill all vectors
 
       Res   : C.int;
