Description: Fix ICE in symtab.c
Origin: https://gcc.gnu.org/viewcvs/gcc?view=revision&revision=220983
Author: Kai Tietz <kai.tietz@onevision.com>

--- branches/gcc-4_9-branch/gcc/symtab.c	2015/02/25 16:44:26	220982
+++ branches/gcc-4_9-branch/gcc/symtab.c	2015/02/25 16:46:34	220983
@@ -969,6 +969,7 @@
   DECL_VISIBILITY_SPECIFIED (decl) = 0;
   DECL_VISIBILITY (decl) = VISIBILITY_DEFAULT;
   TREE_PUBLIC (decl) = 0;
+  DECL_DLLIMPORT_P (decl) = 0;
   if (!DECL_RTL_SET_P (decl))
     return;
 
@@ -1222,6 +1223,7 @@
 
   /* Otherwise create a new one.  */
   new_decl = copy_node (node->decl);
+  DECL_DLLIMPORT_P (new_decl) = 0;
   DECL_NAME (new_decl) = clone_function_name (node->decl, "localalias");
   if (TREE_CODE (new_decl) == FUNCTION_DECL)
     DECL_STRUCT_FUNCTION (new_decl) = NULL;
